### PR TITLE
Bratseth/document parsing error messages

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Comment by bratseth** [comment]
@geirst please review. The changes are in the first commit.

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java [hunk 1]
```diff
@@ -1400,23 +1404,27 @@ public void tensor_modify_update_treats_the_input_tensor_as_sparse() {
...
-                "  'fields': {",
-                "    'something': {",
-                "      'modify': {} }}}"));
-        reader.readSingleDocument(DocumentParser.SupportedOperation.UPDATE, "id:unittest:smoke::doc1");
+        try {
+            JsonReader reader = createReader(inputJson("{ 'update': 'id:unittest:smoke::doc1',",
+                                                       "  'fields': {",
+                                                       "    'something': {",
+                                                       "      'modify': {} }}}"));
+            reader.readSingleDocument(DocumentParser.SupportedOperation.UPDATE, "id:unittest:smoke::doc1");
```
Comment (geirst) [inline document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java]: We should fail on the next line if the exception was not thrown.

**Review by geirst — COMMENTED** [review]
Overall looks good, one comment to look at

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java (hunk 1 above)
Comment (bratseth) [reply document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java]: Good catch - thanks

**Review by geirst — APPROVED** [review]
👍

**Comment by bratseth** [comment]
Please mr. Merge Stop Enforcer

**Pull request merged**